pkg/descheduler/evictions/evictions.go

```diff
@@ -97,19 +98,21 @@ func (pe *PodEvictor) TotalEvicted() int {
 // EvictPod returns non-nil error only when evicting a pod on a node is not
 // possible (due to maxPodsToEvictPerNode constraint). Success is true when the pod
 // is evicted on the server side.
-func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node, reasons ...string) (bool, error) {
+func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node, strategy string, reasons ...string) (bool, error) {
 	var reason string
 	if len(reasons) > 0 {
 		reason = " (" + strings.Join(reasons, ", ") + ")"
```

Comment:
Continuing from what we're talking about above (https://github.com/kubernetes-sigs/descheduler/pull/505/files#r584629100), maybe we can change this line to:
```go
		reason = strategy + " (" + strings.Join(reasons, ", ") + ")"
```
then,  all the calls to `EvictPod` can be shortened to just 1 parameter (so, no longer passing in the optional `reasons` list, but still keeping that available):
```go
podEvictor.EvictPod(ctx, pod, nodeMap[nodeName], "RemoveDuplicatePods")
```

this way, we don't have to remove the `reasons` parameter but we clean up those weird-looking function calls. We can then start some work to actually use `reasons` for each strategy in a more informative way than how it is currently. What do you think?

Reply:
Good idea!!! Updated.